solc/CommandLineInterface.cpp

```diff
@@ -1164,14 +1168,14 @@ void CommandLineInterface::outputCompilationResults()
 	// do we need AST output?
 	handleAst();
 
-	if (
-		!m_compiler->compilationSuccessful() &&
-		m_options.output.stopAfter == CompilerStack::State::CompilationSuccessful
-	)
-	{
-		serr() << endl << "Compilation halted after AST generation due to errors." << endl;
+	CompilerOutputs astOutputSelection;
+	astOutputSelection.astCompactJson = true;
+	if (m_options.compiler.outputs == CompilerOutputs() || m_options.compiler.outputs == astOutputSelection)
 		return;
-	}
```

Comment:
This is actually a minor bugfix. Because of this check, `--error-recovery` on the CLI used to always exit without an error code, even when there were errors. It maybe made sense if AST was requested (we produce output after all) but this happened regardless of that. Now it properly exits with an error, which can be seen in tests.